.github/workflows/auto-update-dependencies.yml

```diff
@@ -9,7 +9,7 @@ permissions:
   contents: read
 
 env:
-  BRANCH_NAME: ${{ github.head_ref }}
+  BRANCH_NAME: ${{ github.ref_name }}
```

Comment:
Note: the value of `head_ref` is an empty string when it's accessed via a direct push to the branch. It is only accessible when doing it via a pull request which is why a previous PR related to this functionality passed the testing. I don't think this was ever actually posting any commits to `develop`, though.
Setting the value to `ref_name` instead will allow us to access that `develop` value which is what we want.